resources/dw/virtualwarehouse/hive/model_hive_vw.go

```diff
@@ -11,21 +11,151 @@
+	vwType := models.VwType("hive")
+	return &models.CreateVwRequest{
+		ClusterID:             p.ClusterID.ValueStringPointer(),
+		DbcID:                 p.DatabaseCatalogID.ValueStringPointer(),
+		EbsLLAPSpillGB:        p.getEbsLLAPSpillGB(),
+		ImageVersion:          p.getImageVersion(),
+		Name:                  p.Name.ValueStringPointer(),
+		NodeCount:             utils.Int64To32(p.GroupSize),
+		PlatformJwtAuth:       p.PlatformJwtAuth.ValueBoolPointer(),
+		QueryIsolationOptions: p.getQueryIsolationOptions(),
+		Autoscaling:           p.getAutoscaling(),
+		AvailabilityZone:      p.getAvailabilityZone(),
+		Tags:                  p.getTags(),
+		Config:                p.getServiceConfig(),
+		VwType:                &vwType,
+	}
+}
+
+func (p *resourceModel) getImageVersion() string {
+	if p.ImageVersion.IsNull() || p.ImageVersion.IsUnknown() || p.ImageVersion.String() == "<unknown>" {
+		return ""
+	}
+	return strings.TrimPrefix(strings.TrimSuffix(p.ImageVersion.String(), "\""), "\"")
+}
+
+func (p *resourceModel) getServiceConfig() *models.ServiceConfigReq {
+	return &models.ServiceConfigReq{
+		ApplicationConfigs: nil,
+		CommonConfigs:      nil,
+		EnableSSO:          p.EnableSSO.ValueBool(),
+		LdapGroups:         utils.FromListValueToStringList(p.LdapGroups),
+	}
+}
+
+func (p *resourceModel) getTags() []*models.TagRequest {
+	if p.AwsOptions.Tags.IsNull() {
+		return nil
+	}
+	var tags []*models.TagRequest
+	for k, v := range p.AwsOptions.Tags.Elements() {
+		if v.IsNull() {
+			continue
+		}
+		value := strings.TrimPrefix(strings.TrimSuffix(v.String(), "\""), "\"")
+		tags = append(tags, &models.TagRequest{
+			Key:   &k,
+			Value: &value,
+		})
+	}
+	return tags
+}
+
+func (p *resourceModel) getQueryIsolationOptions() *models.QueryIsolationOptionsRequest {
+	return &models.QueryIsolationOptionsRequest{
+		MaxQueries:       utils.Int64To32(p.MaxConcurrentIsolatedQueries),
+		MaxNodesPerQuery: utils.Int64To32(p.MaxNodesPerIsolatedQuery),
+	}
+}
+
+func (p *resourceModel) getAvailabilityZone() string {
+	if p.AwsOptions == nil {
+		return ""
+	}
+	return p.AwsOptions.AvailabilityZone.ValueString()
+}
+
+func (p *resourceModel) getEbsLLAPSpillGB() int32 {
+	if p.AwsOptions == nil {
+		return 0
+	}
+	return utils.Int64To32(p.AwsOptions.EbsLLAPSpillGb)
+}
+
+func (p *resourceModel) getAutoscaling() *models.AutoscalingOptionsCreateRequest {
+	return &models.AutoscalingOptionsCreateRequest{
+		MinClusters:               utils.Int64To32Pointer(p.MinGroupCount),
+		MaxClusters:               utils.Int64To32Pointer(p.MaxGroupCount),
+		DisableAutoSuspend:        p.DisableAutoSuspend.ValueBool(),
+		AutoSuspendTimeoutSeconds: utils.Int64To32(p.AutoSuspendTimeoutSeconds),
+		HiveScaleWaitTimeSeconds:  utils.Int64To32(p.ScaleWaitTimeSeconds),
+		HiveDesiredFreeCapacity:   utils.Int64To32(p.Headroom),
+	}
+}
+
+func (p *resourceModel) setFromDescribeVwResponse(resp *models.DescribeVwResponse) {
+	p.ID = types.StringValue(resp.Vw.ID)
```

Comment:
Oh if something goes wrong, error is already handled when we get the Vw details, if the item is not found, that is an error too. I don't think there is a need to play safe here because we already should be safe. It is more likely that some data is not populated in the response, like the Endpoints, but at least the hue one should be always populated.

docs says: 
```
DescribeVwOK describes a response with status code 200, with default header values.
```
If you don't mind I would skip this check here

Reply:
anyway, I will add a return early in case of nil